backend/game/dice_handler.py

```diff
@@ -0,0 +1,25 @@
+import game.dice.dice as dice
+
+
+class DiceHandler:
+    def __init__(self):
+        self.dice_values = []
+        self.re_rolls_left = 0
+
+    def start_turn(self, starting_dice_count, re_roll_count):
+        self.dice_values = dice.roll_many(starting_dice_count)
+        self.re_rolls_left = re_roll_count
+
+    def re_roll_dice(self, indexes_of_dice_to_re_roll):
+        if self.re_rolls_left > 0:
+            try:
+                for val in indexes_of_dice_to_re_roll:
+                    self.dice_values[val] = dice.roll()
+                self.re_rolls_left -= 1
```

Comment:
With it like this, you could potentially get SOME of the dice rerolled and then have the last one be an index error or type error to not have the reroll counter decrement.

Reply:
Ooh so basically pass reroll [ 3,4,5,19] and you get free re-rolls. Good point.

Reply:
updated method to ignore bad indexes